domain/application/service/status.go

```diff
@@ -71,13 +71,36 @@ func encodeUnitWorkloadStatusType(s status.Status) (application.UnitWorkloadStat
 	}
 }
 
+// decodeUnitWorkloadStatusType converts a db unit workload status id to a core.
+// Implicitly validates the status type.
+func decodeUnitWorkloadStatusType(s application.UnitWorkloadStatusType) (status.Status, error) {
+	switch s {
+	case application.UnitWorkloadStatusUnset:
+		return status.Unset, nil
+	case application.UnitWorkloadStatusUnknown:
+		return status.Unknown, nil
+	case application.UnitWorkloadStatusMaintenance:
+		return status.Maintenance, nil
+	case application.UnitWorkloadStatusWaiting:
+		return status.Waiting, nil
+	case application.UnitWorkloadStatusBlocked:
+		return status.Blocked, nil
+	case application.UnitWorkloadStatusActive:
+		return status.Active, nil
+	case application.UnitWorkloadStatusTerminated:
+		return status.Terminated, nil
+	default:
+		return "", errors.Errorf("unknown workload status %q", s)
+	}
```

Comment:
Add a test for this i.e. ensure that an encoded type can be decoded etc.

See: https://github.com/juju/juju/blob/main/domain/machine/service/status_test.go#L80-L87

Reply:
Done